x-pack/plugins/lens/public/shared_components/coloring/color_ranges/color_ranges_validation.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import { isValidColor } from '../utils';
+
+import type { ColorRange, ColorRangeAccessor } from './types';
+
+/** @internal **/
+type ColorRangeValidationErrors = 'invalidColor' | 'invalidValue' | 'greaterThanMaxValue';
+
+/** @internal **/
+export interface ColorRangeValidation {
+  errors: ColorRangeValidationErrors[];
+  isValid: boolean;
+}
+
+/** @internal **/
+export const getErrorMessages = (colorRangesValidity: Record<string, ColorRangeValidation>) => {
+  return [
+    ...new Set(
+      Object.values(colorRangesValidity)
+        .reduce<ColorRangeValidationErrors[]>((acc, item) => [...acc, ...item.errors], [])
+        .flat()
```

Comment:
Would a `map` suffice here instead of a `reduce`?

```suggestion
    ...new Set(
      Object.values(colorRangesValidity)
        .map((item) => item.errors)
        .flat()
        )
```